### DISCUSSION TITLE
OpenTelemetryDependencyInjectionMeterProviderBuilderExtensions.AddInstrumentation Usage

### DISCUSSION THREAD

**Discussion by gokceng-intertech** [open-telemetry/opentelemetry-dotnet#5209] [General]
Hi,
I've checked the repo (searching by `AddInstrumentation` usages in Github) to understand what might be the cases or how I can use `AddInstrumentation` method but couldn't find anything.
Is there a specific interface that added instrument needs to implement or is there any example of using this method?
Thank you.

**Comment by cijothomas** [comment]
`AddInstrumentation` is to be used by people authoring instrumentation libraries.

**Reply by gokceng-intertech** [reply]
Thanks @cijothomas , is there any library or framework or such that I can look at to understand the specifics?

**Reply by cijothomas** [reply]
https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/docs/trace/extending-the-sdk#instrumentation-library The doc has links to example implementations. But I am not sure why you need to understand that... Are you writing instrumentation library ? If yes, could you share which library are you writing instrumentation for?
